File: SDKs/twilio.com/twilio_verify_v2/1.36.0/go/pkg/models/operations/listwebhook.go
package operations

import (
	"openapi/pkg/models/shared"
)

var ListWebhookServers = []string{
	"https://verify.twilio.com",
}

type ListWebhookPathParams struct {
	ServiceSid string `pathParam:"style=simple,explode=false,name=ServiceSid"`
}

type ListWebhookQueryParams struct {
	PageSize *int64 `queryParam:"style=form,explode=true,name=PageSize"`
}

type ListWebhookSecurity struct {
	AccountSidAuthToken shared.SchemeAccountSidAuthToken `security:"scheme,type=http,subtype=basic"`
}

type ListWebhookRequest struct {
	ServerURL   *string
	PathParams  ListWebhookPathParams
	QueryParams ListWebhookQueryParams
	Security    ListWebhookSecurity
}

type ListWebhookListWebhookResponseMeta struct {
	FirstPageURL    *string `json:"first_page_url,omitempty"`
	Key             *string `json:"key,omitempty"`
	NextPageURL     *string `json:"next_page_url,omitempty"`
	Page            *int64  `json:"page,omitempty"`
	PageSize        *int64  `json:"page_size,omitempty"`
	PreviousPageURL *string `json:"previous_page_url,omitempty"`
	URL             *string `json:"url,omitempty"`
}

type ListWebhookListWebhookResponse struct {
	Meta     *ListWebhookListWebhookResponseMeta `json:"meta,omitempty"`
	Webhooks []shared.VerifyV2ServiceWebhook     `json:"webhooks,omitempty"`
}

type ListWebhookResponse struct {
	ContentType         string
	ListWebhookResponse *ListWebhookListWebhookResponse
	StatusCode          int64
}

File: SDKs/twilio.com/twilio_verify_v2/1.37.3/typescript/src/sdk/models/operations/listwebhook.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";

export const ListWebhookServerList = [
	"https://verify.twilio.com",
] as const;


export class ListWebhookPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=ServiceSid" })
  serviceSid: string;
}


export class ListWebhookQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=PageSize" })
  pageSize?: number;
}


export class ListWebhookSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=http;subtype=basic" })
  accountSidAuthToken: shared.SchemeAccountSidAuthToken;
}


export class ListWebhookListWebhookResponseMeta extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=first_page_url" })
  firstPageUrl?: string;

  @SpeakeasyMetadata({ data: "json, name=key" })
  key?: string;

  @SpeakeasyMetadata({ data: "json, name=next_page_url" })
  nextPageUrl?: string;

  @SpeakeasyMetadata({ data: "json, name=page" })
  page?: number;

  @SpeakeasyMetadata({ data: "json, name=page_size" })
  pageSize?: number;

  @SpeakeasyMetadata({ data: "json, name=previous_page_url" })
  previousPageUrl?: string;

  @SpeakeasyMetadata({ data: "json, name=url" })
  url?: string;
}


export class ListWebhookListWebhookResponse extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=meta" })
  meta?: ListWebhookListWebhookResponseMeta;

  @SpeakeasyMetadata({ data: "json, name=webhooks", elemType: shared.VerifyV2ServiceWebhook })
  webhooks?: shared.VerifyV2ServiceWebhook[];
}


export class ListWebhookRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  serverUrl?: string;

  @SpeakeasyMetadata()
  pathParams: ListWebhookPathParams;

  @SpeakeasyMetadata()
  queryParams: ListWebhookQueryParams;

  @SpeakeasyMetadata()
  security: ListWebhookSecurity;
}


export class ListWebhookResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  listWebhookResponse?: ListWebhookListWebhookResponse;

  @SpeakeasyMetadata()
  statusCode: number;
}
